Sources/Sync/SyncCoordinator/Participants/CampaignSyncParticipant.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  CampaignSyncParticipant.swift
+//  RoverSync
+//
+//  Created by Andrew Clunis on 2019-01-22.
+//  Copyright © 2019 Rover Labs Inc. All rights reserved.
+//
+
+import Foundation
+import os.log
+import CoreData
+
+class CampaignSyncParticipant: PagingSyncParticipant {
+    typealias Response = CampaignsSyncResponse
+    
+    let context: NSManagedObjectContext
+    let userDefaults: UserDefaults
+    
+    var cursorKey: String {
+        return "io.rover.RoverSync.campaignsCursor"
+    }
+    
+    init(context: NSManagedObjectContext, userDefaults: UserDefaults) {
+        self.context = context
+        self.userDefaults = userDefaults
+    }
+    
+    func nextRequest(cursor: String?) -> SyncRequest {
+        let orderBy: [String: Any] = [
+            "field": "UPDATED_AT",
+            "direction": "ASC"
+        ]
+        
+        var values: [String: Any] = [
+            "first": 10, // TODO: sane page size?
```

Comment:
My gut says this can be much larger. Maybe park it for now but create a GitHub issue to determine average campaign size (when Gzipped) and come up with a number based on data.

Reply:
https://github.com/RoverPlatform/rover-ios/issues/403